tools/docker-compose-datasets-server.yml

```diff
@@ -69,6 +69,40 @@ services:
     depends_on:
       - mongodb
     restart: unless-stopped
+  worker-config-names:
+    # build:
+    #   context: ..
+    #   dockerfile: workers/datasets_based/Dockerfile
+    image: ${IMAGE_WORKER_DATASETS_BASED?IMAGE_WORKER_DATASETS_BASED env var must be provided}
+    volumes:
+      - splits-datasets-cache:${HF_DATASETS_CACHE-/datasets-cache}:rw
+      - splits-modules-cache:${HF_DATASETS_CACHE-/modules-cache}:rw
+      - splits-numba-cache:${NUMBA_CACHE_DIR-/numba-cache}:rw
+    extends:
+      file: docker-compose-base.yml
+      service: datasets-worker
+    environment:
+      DATASETS_BASED_ENDPOINT: "/config-names" # hard-coded
+    depends_on:
+      - mongodb
+    restart: always
+  worker-split-names:
+    # build:
+    #   context: ..
+    #   dockerfile: workers/datasets_based/Dockerfile
+    image: ${IMAGE_WORKER_DATASETS_BASED?IMAGE_WORKER_DATASETS_BASED env var must be provided}
+    volumes:
+      - splits-datasets-cache:${HF_DATASETS_CACHE-/datasets-cache}:rw
+      - splits-modules-cache:${HF_DATASETS_CACHE-/modules-cache}:rw
+      - splits-numba-cache:${NUMBA_CACHE_DIR-/numba-cache}:rw
+    extends:
+      file: docker-compose-base.yml
+      service: datasets-worker
+    environment:
+      DATASETS_BASED_ENDPOINT: "/split-names" # hard-coded
+    depends_on:
+      - mongodb
+    restart: always
```

Comment:
we can use this for a local dev env right ?

Reply:
yes, you can launch `make start` at the root, and it will use this docker compose file to deploy all the images.

Until now, I've not really used a "local dev env":
- if I want to try something, I generally create a test (be it a unit test if it only affects a service or a worker, or an e2e test if I want to test the relation between them)
- alternately, I just go to one of the services or worker, launch `poetry run python` then interact with the prompt to test the methods.
- if I need to test the whole app manually, I deploy on the ephemeral k8s cluster

But now that we are various people working on the project, the last point is not really a good idea anymore, so, it would be better to launch locally when one wants to test the whole app manually.